### ISSUE TITLE
Specifying of SMSC GW GT and TT (TranslationType) by mproc rules

### ISSUE THREAD

**Issue by vetss** [RestComm/smscgateway#252]
1) Add int Sms class two extra fields that must be persisted in a database (defaul values are null)
- String ServiceCenterGT
- Integer ServiceCenterTT 

2) PostArrivalProcessor - add actions 
- updateServiceCenterGT(String val);
- updateRemoteTT(Integer val);

3) When SRI and MT message sending we need to use specified GT string / TT (if they were updated by mproc rules).

4) add extra actions into defaul mproc rules that will set values
mtlocalsccpgt ****
mtremotesccptt ****

5) we need to update a manual - chapter http://documentation.telestax.com/core/smsc/SMSC_Gateway_Admin_Guide.html#_mproc_rules_fundamentals

**Comment by faizann** [comment]
Hi Sergey

I have picked this one up.

**Comment by vetss** [comment]
Hello @faizann 

please hold on,, @olenara is processing the issue. Sorry, I have only today found your message.

**Comment by faizann** [comment]
Roger that. I have done changes but I will take olenara.
We also have seen cases where SCCP CallingGT has to be set and is different from ServiceCenter GT. This happens when ServiceCenterGT is used to provide more info about the originating operator by adding MNC/MCC suffix. This is called OC Compliance in GSMA standard. We can add another param only for sccpCallingGt that overrides serviceCenterGt if found to be set else serviceCenterGt is used in both SCCP and MAP layers.

**Comment by vetss** [comment]
@faizann 

1) is it for MT messages only (or affects also MO/HR) ?
2) if MT only - can we specify a pair of values (ServiceCenter GT and SCCP CallingGT) for each networkID value ? Or we need to specify several possible SCCP CallingGT for same ServiceCenter GT and we need mproc rules action for it (like "mtlocalsccpgt ****").

**Comment by faizann** [comment]
Hi @vetss 

It is important for HR/MO as well. The OC Compliance is needed when outbound SRI/MTFSM are sent in correspondence to inbound and we want to relay the originator by appending MCC/MNC to the ServiceCenterGT.  Hence a different callingGT for SCCP and for ServiceCenterGT.
If you agree then I can pick this one up.
NetworkID unfortunately won't be scalable as then we need to configure a lot of things and MProc can't hold the logic.

**Comment by vetss** [comment]
Hello @faizann 

it will be very good if you can cover the issue of "different callingGT for SCCP and for ServiceCenterGT". 
Can you open a separate issue with a requirenments description details. We can discuss then the best solution for it.
We have to close this issue soon because it is for some concrete case and better to have a discussion in a separate issue.

**Comment by vetss** [comment]
After discussing with a customer we have a change for this issue task:
We need to update for mproc rules:
a) GT digits for CallingPartyAdrress (origination address)
b) TT for CalledPartyAddress (destination address)

because of it we need to rename CLI names:
mtlocalgt ****
mtremotett ****
and update code accordingly

We need also to update SMSC GW manual http://documentation.telestax.com/core/smsc/SMSC_Gateway_Admin_Guide.html#_mproc_rules_fundamentals for new commands

I will update also a first record of the issue

**Comment by faizann** [comment]
Hi @vetss 
Why not also have mtlocaltt and mtremotegt since we are doing it anyway.

**Comment by faizann** [comment]
This is the new issue https://github.com/RestComm/smscgateway/issues/258

**Comment by vetss** [comment]
Hi @faizann 

> Why not also have mtlocaltt and mtremotegt since we are doing it anyway.

We are doing this issue for some customer and therefor we need to have hier only a related staff. Sorry for some messing.

**Comment by vetss** [comment]
Changing of CLI parametrs :
mtlocalgt **** -> mtlocalsccpgt
mtremotett **** -> mtremotesccptt

**Comment by faizann** [comment]
Is this issue finished or something is pending on this one?

**Comment by vetss** [comment]
About finished, we need one little update, @faizann

**Comment by vetss** [comment]
@faizann I think we have finished this issue and code is in master branch.
We will test in a customer side, may be some extra feedback / fix.

**Comment by faizann** [comment]
Hi
2 issues in the commits:

1. RSDS
I think for consistency in RsdsSbb the mtlocalsccpgt should be used for servicecenter address.
This applies for createNewDialog call as well as servicecenter address put into MAP layer as that is used for notification.
This is the line where we need to overriding.
https://github.com/RestComm/smscgateway/blob/master/core/slee/services/sbbs/src/main/java/org/mobicents/smsc/slee/services/mt/RsdsSbb.java#L332

2. Documentation
Also in the documentation on this line it states Destination Global Title will be changed but it should be Source Global Title that gets affected.
https://github.com/RestComm/smscgateway/blob/b0c7aadea2e7c8fa01032bf6f229fa4a098319a1/management/ui-management/src/main/webapp/modules/mproc.html#L411

**Comment by vetss** [comment]
Updated fixing list:

https://github.com/RestComm/smscgateway/commit/18a9536315ea123dc78912559f170e148f7dff06
https://github.com/RestComm/smscgateway/commit/5d97c6061e871810b7349f116a8d241213b9bef5
https://github.com/RestComm/smscgateway/commit/baf14292754869191f402cdbc7a6be86642ca27c
https://github.com/RestComm/smscgateway/commit/107150b4766a455d14ada8ed418bc470c5c06507
https://github.com/RestComm/smscgateway/commit/db8d87d3c557bd3827d73041ef39431765fbf02c
https://github.com/RestComm/smscgateway/commit/b735e23bed78c3dbe5fe562d5cc5e23c0ba220bd
https://github.com/RestComm/smscgateway/commit/563f9b575e1a249c3a41b522a684bc9cfd69d218

**Comment by vetss** [comment]
Hello @faizann 

Thanks for valuable bugs mentioning. RSDS update is reasonable.
I added a fix for it.
